tokio/src/task/error.rs

```diff
@@ -46,3 +47,15 @@ impl fmt::Debug for JoinError {
 }
 
 impl std::error::Error for JoinError {}
+
+impl From<JoinError> for io::Error {
+    fn from(src: JoinError) -> io::Error {
+        io::Error::new(
+            io::ErrorKind::Other,
+            match src.repr {
+                Repr::Cancelled => "task was cancelled",
+                Repr::Panic(_) => "task panicked",
+            },
```

Comment:
would it be better to just say 
```rust
 io::Error::new(io::ErrorKind::Other, src)
```
instead, here? 

that way, code like
```rust
if error.source().iter().contains(|e| e.is::<tokio::task::JoinError>()) {
    // ...
}
```

Reply:
I don't have a strong opinion. It would force an allocation. If you want to pursue this, can you open a follow up issue to discuss there?

Reply:
will do. we can always change it later!